packages/shared/src/hooks/feed/useCardCover.tsx

```diff
@@ -28,9 +28,21 @@ export const useCardCover = ({
 }: UseCardCoverProps): UseCardCover => {
   const { shouldShowOverlay, onInteract } = usePostShareLoop(post);
   const shouldShowReminder = useBookmarkReminderCover(post);
+  const [interaction, setInteraction] = useState<'upvote' | 'bookmark' | null>(null);
+  useEffect(() => {
+    if (shouldShowOverlay) {
+      setInteraction('upvote');
+    }
+  }, [shouldShowOverlay]);
+
+  useEffect(() => {
+    if (shouldShowReminder) {
+      setInteraction('bookmark');
+    }
+  }, [shouldShowReminder]);
```

Comment:
Looks amazing to see this get fixed. Although not a big fan of use state/use effect combinations. Can we maybe check within the `usePostShareLoop` if we can incorporate the changes there about tracking the current interaction?